src/main/java/com/coveros/selenified/element/check/Contains.java

```diff
@@ -0,0 +1,281 @@
+     * the element isn't present, this will constitute a failure, same as a
+     * mismatch. This information will be logged and recorded, with a screenshot
+     * for traceability and added debugging support.
+     *
+     * @param expectedText the expected value of the element
+     */
+    void text(String expectedText);
+
+    default String checkText(String expectedText, double waitFor, double timeTook) {
+        // record the action
+        getOutputFile().recordAction(getElement().prettyOutput() + HASTEXT + expectedText + "</b>", waitFor);
+        // get the value
+        String elementValue = getElement().get().text();
+        // record the result
+        if (elementValue == null || !elementValue.contains(expectedText)) {
+            getOutputFile().recordActual(getElement().prettyOutputStart() + TEXT + elementValue + "</b>", timeTook, Success.FAIL);
+        } else {
+            getOutputFile().recordActual(getElement().prettyOutputStart() + TEXT + elementValue + "</b>", timeTook, Success.PASS);
+        }
+        return elementValue;
+    }
+
+    /**
+     * Verifies that the element's value contains the provided expected value.
+     * If the element isn't present or an input, this will constitute a failure,
+     * same as a mismatch. This information will be logged and recorded, with a
+     * screenshot for traceability and added debugging support.
+     *
+     * @param expectedValue the expected value of the element
+     */
+    void value(String expectedValue);
+
+    default String checkValue(String expectedValue, double waitFor, double timeTook) {
+        // record the action and get the attributes
+        String elementValue = getValue(expectedValue, HASVALUE, waitFor);
+        // record the expected
+        if (elementValue == null || !elementValue.contains(expectedValue)) {
+            getOutputFile().recordActual(getElement().prettyOutputStart() + VALUE + elementValue + "</b>", timeTook, Success.FAIL);
+        } else {
+            getOutputFile().recordActual(getElement().prettyOutputStart() + VALUE + elementValue + "</b>", timeTook, Success.PASS);
+        }
+        return elementValue;
+    }
+
+    /**
+     * Verifies that the element's options contains the provided expected
+     * option. If the element isn't present or a select, this will constitute a
+     * failure, same as a mismatch. This information will be logged and
+     * recorded, with a screenshot for traceability and added debugging support.
+     *
+     * @param expectedOption the option expected in the list
+     */
+    void selectOption(String expectedOption);
+
+    default String[] checkSelectOption(String expectedOption, double waitFor, double timeTook) {
+        // record the action, and check for select
+        if (!isPresentSelect(getElement().prettyOutput() + " with the option <b>" + expectedOption +
+                "</b> available to be selected on the page", waitFor)) {
+            return null;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Return an empty array instead of null. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/sonar/coding_rules#rule_key=squid%3AS1168)